snuba/snuba_migrations/functions/0004_functions_v2.py

```diff
@@ -0,0 +1,232 @@
+    Column("examples", AggregateFunction("groupUniqArray(5)", [UUID()])),
+]
+
+
+new_columns: Sequence[NewColumn] = [
+    NewColumn(
+        Column(
+            "worst_v2",
+            AggregateFunction(
+                "argMax",
+                [
+                    Tuple(
+                        (
+                            UUID(),  # profile_id
+                            String(),  # thread_id
+                            DateTime64(  # start_timestamp
+                                precision=6,
+                                modifiers=Modifiers(nullable=True),
+                            ),
+                            DateTime64(  # end_timestamp
+                                precision=6,
+                                modifiers=Modifiers(nullable=True),
+                            ),
+                        )
+                    ),  # end Tuple
+                    Float(64),
+                ],
+            ),
+        ),
+        after="worst",
+    ),  # end new column
+    NewColumn(
+        Column(
+            "examples_v2",
+            AggregateFunction(
+                "groupUniqArray(5)",
+                [
+                    Tuple(
+                        (
+                            UUID(),  # profile_id
+                            String(),  # thread_id
+                            DateTime64(
+                                precision=6,
+                                modifiers=Modifiers(nullable=True),
+                            ),  # start_timestamp
+                            DateTime64(
+                                precision=6,
+                                modifiers=Modifiers(nullable=True),
+                            ),  # end_timestamp
+                        )
+                    ),  # end Tuple
+                ],
+            ),
+        ),
+        after="examples",
+    ),
+    NewColumn(
+        column=Column(
+            "profiling_type",
+            String(Modifiers(low_cardinality=True, default="'transaction'")),
+        ),
+        after="is_application",
+    ),
+]
+
+all_columns = columns + [col.column for col in new_columns]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    data_granularity = 60 * 60  # 1 hour buckets
+    storage_set = StorageSetKey.FUNCTIONS
+
+    local_raw_table = "functions_raw_local"
+    dist_raw_table = "functions_raw_dist"
+
+    local_materialized_table = "functions_mv_local"
+    dist_materialized_table = "functions_mv_dist"
+
+    local_view_table = "functions_v2_local"  # newer materialized
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=self.storage_set,
+                table_name=table_name,
+                column=new_column.column,
+                after=new_column.after,
+                target=target,
+            )
```

Comment:
What's this? Dont most if not all these columns already exist on the table?

Reply:
@Zylphrex I'm only adding 3 missing columns to the existing materialized table (namely `worst_v2`, `example_v2`, `profiling_type`). You can see I'm only adding columns from _new_columns_ `column=new_column.column`

All the other columns that I've listed, it's because I have to specify them in the new `view` (`columns=all_columns`), but I'm not adding all of those again in the materialized functions table.

Reply:
ahh i see 👍